File: shared/src/main/java/com/housemate/shared/dto/chore/request/ChoreAssignmentFilterRequestDTO.java
package com.housemate.shared.dto.chore.request;

import com.housemate.shared.enums.ChoreStatus;
import com.housemate.shared.utils.types.DateRange;
import jakarta.validation.constraints.NotNull;

import java.util.List;
import java.util.UUID;

public record ChoreAssignmentFilterRequestDTO(
        List<ChoreStatus> statuses,
        UUID assigneeId,
        String descriptionContains,

        @NotNull(message = "Date range cannot be null")
        DateRange dateRange
) {}

File: mobile/lib/shared/dto/chore/request/chore_assignment_filter_request_dto.dart
import 'package:json_annotation/json_annotation.dart';
import '../../../enums/chore_status.dart';
import '../../../utils/types/date_range.dart';

part 'chore_assignment_filter_request_dto.g.dart';

@JsonSerializable(explicitToJson: true)
class ChoreAssignmentFilterRequestDTO {
  final List<ChoreStatus>? statuses;
  final String? assigneeId;
  final String? descriptionContains;
  final DateRange dateRange;

  ChoreAssignmentFilterRequestDTO({
    this.statuses,
    this.assigneeId,
    this.descriptionContains,
    required this.dateRange,
  });

  factory ChoreAssignmentFilterRequestDTO.fromJson(Map<String, dynamic> json) =>
      _$ChoreAssignmentFilterRequestDTOFromJson(json);

  Map<String, dynamic> toJson() => _$ChoreAssignmentFilterRequestDTOToJson(this);
}
